HaplotypeMap/BuildHaplotypeMap.wdl

```diff
@@ -0,0 +1,167 @@
+# BuildHaplotypeMap
+#
+# This workflow creates a haplotype map file containing a set of low LD sites for fingerprinting
+#
+# Inputs:
+#
+# 1. Genotyped multisample GVCF
+# 2. Header file. Standard SAM header (should match header of files to be fingerprinted)
+# 3. Pruning parameters, see https://www.cog-genomics.org/plink/2.0/ld for more info. 
+#    In general, higher values for prune window and prune slide result in more pruning, while lower values for prune cutoff and maf result in less pruning
+#  
+#   
+#
+
+task vcftools {
+    String intermediates_prefix
+    File input_vcf
+    
+    command <<<
+  	vcftools --vcf ${input_vcf} --max-missing-count 0 --remove-indels --recode --recode-INFO-all --out ${intermediates_prefix}
+    >>>
+	
+    runtime {
+    disks: "local-disk 1000 HDD"
+    memory: "3500 MB"
+    docker: "jeltje/vcftools"
+  }
+
+	output {
+    File recode_vcf = "${intermediates_prefix}.recode.vcf"
+  }
+}
+
+task plink {
+  File input_vcf
+  String intermediates_prefix
+  String output_prefix
+  Int prune_window
+  Int prune_slide
+  Float prune_cutoff
+  Float min_maf
+  
+  String set_missing = "@:#"
+
+  command <<<
+  	plink --vcf ${input_vcf} --snps-only --maf ${min_maf} --biallelic-only --set-missing-var-ids ${set_missing} --indep-pairwise ${prune_window} ${prune_slide} ${prune_cutoff} --out ${intermediates_prefix}
+    >>>
+
+  runtime {
+    disks: "local-disk 1000 HDD"
+    memory: "3500 MB"
+    docker: "jrose77/plinkdocker"
+  }
+
+  output {
+    File prune_in = "${intermediates_prefix}.prune.in"
+  }
+}
+
+task reformat{
+	String output_prefix
+	File prune_in
+    File recode_vcf
+	File header_file
+
+	command <<<
+python <<CODE
+import pandas as pd
+import vcf
+
+
+vcf_file = "${recode_vcf}"
+prune_in = "${prune_in}"
+header = open("${header_file}")
+new_map = open("${output_prefix}.hapmap.txt", 'w')
+vcf_reader = vcf.Reader(open(vcf_file, 'r', encoding='utf-8'))
+vcf_chr = False
+first = next(vcf_reader)
+if str(first.CHROM)[0:3] == 'chr':
+	vcf_chr = True
+sites_to_include = open(prune_in)
+sites_list = []
```

Comment:
make this a set not a list.  Since you will be checking whether sites are included over and over, you want the check for inclusion to be very fast, but don't care about being able to iterate over the set of included sites particularly efficienty.